presto-main/src/test/java/com/facebook/presto/sql/planner/TestPlanFragmenter.java

```diff
@@ -0,0 +1,135 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.execution.NodeTaskMap;
+import com.facebook.presto.execution.scheduler.LegacyNetworkTopology;
+import com.facebook.presto.execution.scheduler.NodeScheduler;
+import com.facebook.presto.execution.scheduler.NodeSchedulerConfig;
+import com.facebook.presto.execution.scheduler.nodeSelection.NodeSelectionStats;
+import com.facebook.presto.metadata.CatalogManager;
+import com.facebook.presto.metadata.InMemoryNodeManager;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.tpch.TpchConnectorFactory;
+import com.facebook.presto.util.FinalizerService;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.facebook.airlift.testing.Assertions.assertNotEquals;
+import static com.facebook.presto.SystemSessionProperties.FORCE_COORDINATOR_EXECUTION;
+import static com.facebook.presto.spi.WarningCollector.NOOP;
+import static com.facebook.presto.sql.planner.LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED;
+import static com.facebook.presto.sql.planner.SystemPartitioningHandle.COORDINATOR_DISTRIBUTION;
+import static com.facebook.presto.testing.TestingSession.createBogusTestingCatalog;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static org.testng.Assert.assertEquals;
+
+public class TestPlanFragmenter
+{
+    private FinalizerService finalizerService;
+    private NodeScheduler nodeScheduler;
+    private LocalQueryRunner queryRunnerEnableCoordinatorExecution;
+    private LocalQueryRunner queryRunnerDisableCoordinatorExecution;
+
+    @BeforeClass
+    public void setUp()
+    {
+        CatalogManager catalogManager = new CatalogManager();
+        catalogManager.registerCatalog(createBogusTestingCatalog("tpch"));
+
+        finalizerService = new FinalizerService();
+        finalizerService.start();
+        nodeScheduler = new NodeScheduler(
+                new LegacyNetworkTopology(),
+                new InMemoryNodeManager(),
+                new NodeSelectionStats(),
+                new NodeSchedulerConfig().setIncludeCoordinator(true),
+                new NodeTaskMap(finalizerService));
+        Map<String, String> sessionProperties = new HashMap<>();
+        sessionProperties.put(FORCE_COORDINATOR_EXECUTION, "true");
+        queryRunnerDisableCoordinatorExecution = createQueryRunner(ImmutableMap.of());
+        queryRunnerEnableCoordinatorExecution = createQueryRunner(sessionProperties);
```

Comment:
`ImmutableMap.of(FORCE_COORDINATOR_EXECUTION, "true")`

Reply:
Actually you don't need to create two separate executors as you can set session properties on per query basis. Please take a look at `AbstractTestVerboseMemoryExceededErrors`. Also it is generally discouraged creating a query runner manually. We have a higher level framework, `AbstractTestQueryFramework`, for that purpose that implements lifecycle management for a query runner.

Reply:
i fix the test ,but i don`t use the the parent class of `AbstractTestQueryFramework`, because the `presto-test` module depend `presto-main` module which is my test code module.they will circle dependency